### PR TITLE
Changes to accommodate GLOW build using LLVM 10.0

### PR DESCRIPTION
**Pull request by bogdprof**
Summary:
Changes to accommodate GLOW build using LLVM 10.0

Documentation:

[Optional Fixes #issue]

Test Plan:

Please see a detailed explanation of how to fill out the fields in the relevant sections in PULL_REQUEST.md.

### CODE REVIEW

File: lib/Support/Debug.cpp [hunk 1]
```diff
@@ -41,9 +41,11 @@ namespace glow {
 /// Exported boolean set by -debug-glow option.
 bool DebugFlag = false;
 
+#ifndef NDEBUG
```
Comment (opti-mix) [inline lib/Support/Debug.cpp:44]: This is unrelated to the scope of this PR. Please remove.

Comment (bogdprof) [reply lib/Support/Debug.cpp:44]: Actually it is related to LLVM 10.0.
The "isCurrentDebugType()" function needs to implemented only if NDEBUG is not defined. 
Otherwise, there is a macro-definition in "include/llvm/Support/Debug.h" which is expanded (when NDEBUG is defined) and will replace "isCurrentDebugType" with "(false)".

#define isCurrentDebugType(X) (false)

This was not the case before LLVM 10.0

**Comment by bogdprof** [comment]
Any updates on this?
Is there any interest in importing these changes?

**Comment by mciprian13** [comment]
@opti-mix Kind reminder for you tell us whether this PR is ready to get landed. Thanks!
